### PR TITLE
refactor(tests): `assert.Error`->`require.Error` (remainder)

### PR DESCRIPTION
**Pull request by Joibel**
This is [part of a series](https://github.com/argoproj/argo-workflows/pulls?q=is%3Apr+testifylint+) of test tidies started by #13365.

The aim is to enable the testifylint golangci-lint checker.

This commit converts assert.Error checks into require.Error for the rest of the code base.

In some places checks have been coaleced - in particular the pattern

```go
if assert.Error() {
    assert.Contains(..., "message")
}
```

is now
```go
require.ErrorContains(..., "message")
```

Getting this wrong and missing the Contains is still valid go, so that's a mistake I may have made.

### CODE REVIEW

File: test/e2e/cli_test.go [hunk 1]
```diff
@@ -135,47 +131,45 @@ func (s *CLISuite) TestVersion() {
...
+					assert.Contains(t, lines[7], "Platform:")
+					assert.Contains(t, lines[8], "argo-server:")
+					assert.Contains(t, lines[9], "BuildDate:")
+					assert.Contains(t, lines[10], "GitCommit:")
+					assert.Contains(t, lines[11], "GitTreeState:")
+					assert.Contains(t, lines[12], "GitTag:")
+					assert.Contains(t, lines[13], "GoVersion:")
+					assert.Contains(t, lines[14], "Compiler:")
+					assert.Contains(t, lines[15], "Platform:")
 				}
```
Comment (agilgur5) [inline test/e2e/cli_test.go:159]: you changed several below this to `require.Len` as well (as mentioned in https://github.com/argoproj/argo-workflows/pull/13401#discussion_r1696246571), so I'd this as well for consistency. this one's also the longest in terms of lines of indentation changes

```suggestion
				require.Len(t, lines, 17)
				assert.Contains(t, lines[0], "argo:")
				assert.Contains(t, lines[1], "BuildDate:")
				assert.Contains(t, lines[2], "GitCommit:")
				assert.Contains(t, lines[3], "GitTreeState:")
				assert.Contains(t, lines[4], "GitTag:")
				assert.Contains(t, lines[5], "GoVersion:")
				assert.Contains(t, lines[6], "Compiler:")
				assert.Contains(t, lines[7], "Platform:")
				assert.Contains(t, lines[8], "argo-server:")
				assert.Contains(t, lines[9], "BuildDate:")
				assert.Contains(t, lines[10], "GitCommit:")
				assert.Contains(t, lines[11], "GitTreeState:")
				assert.Contains(t, lines[12], "GitTag:")
				assert.Contains(t, lines[13], "GoVersion:")
				assert.Contains(t, lines[14], "Compiler:")
				assert.Contains(t, lines[15], "Platform:")
```

File: test/e2e/cli_test.go [hunk 2]
```diff
@@ -135,47 +131,45 @@ func (s *CLISuite) TestVersion() {
...
-					if assert.Len(t, lines, 3) {
-						assert.Contains(t, lines[0], "argo:")
-						assert.Contains(t, lines[1], "argo-server:")
-					}
+				require.NoError(t, err)
+				lines := strings.Split(output, "\n")
+				if assert.Len(t, lines, 3) {
+					assert.Contains(t, lines[0], "argo:")
+					assert.Contains(t, lines[1], "argo-server:")
 				}
```
Comment (agilgur5) [inline test/e2e/cli_test.go:173]:
```suggestion
				require.Len(t, lines, 3)
				assert.Contains(t, lines[0], "argo:")
				assert.Contains(t, lines[1], "argo-server:")
```

same as above

**Review by agilgur5 — COMMENTED** [review]
I didn't realize there were still some leftover. One question and two small consistency suggestions below, otherwise LGTM

Would you like to tackle the remaining `if assert.*` -> `require.*` changes as mentioned in https://github.com/argoproj/argo-workflows/pull/13401#discussion_r1696246571 etc? the vast majority in my searches were actually `Error`/`NoError`, so there should only be a small number remaining

**Review by agilgur5 — APPROVED** [review]
Made the changes myself above, LGTM now

File: test/e2e/retry_test.go [hunk 3]
```diff
@@ -181,10 +181,10 @@ spec:
 		ctx := context.Background()
 		podLogOptions := &apiv1.PodLogOptions{Container: "c3"}
 		stream, err := s.KubeClient.CoreV1().Pods(ns).GetLogs(name, podLogOptions).Stream(ctx)
-		s.NoError(err)
+		s.Require().NoError(err)
 		defer stream.Close()
 		logBytes, err := io.ReadAll(stream)
-		s.NoError(err)
+		s.Require().NoError(err)
```
[the remark this answers is not in the document]
Comment (Joibel) [reply test/e2e/retry_test.go:187]: None of these assert->require changes are [auto-fixed](https://github.com/Antonboom/testifylint?tab=readme-ov-file#require-error), but it does suggest the fixes if you fix it wrong as part of [suite-dont-use-pkg](https://github.com/Antonboom/testifylint?tab=readme-ov-file#suite-dont-use-pkg) IIRC.

**Comment by Joibel** [comment]
> I didn't realize there were still some leftover.

This is the [first bullet point](https://github.com/argoproj/argo-workflows/pull/13396#issuecomment-2254223469) although there are some here outside `e2e` still, not sure why they were missed before.

> Would you like to tackle the remaining `if assert.*` -> `require.*` changes as mentioned in [#13401 (comment)](https://github.com/argoproj/argo-workflows/pull/13401#discussion_r1696246571) etc? the vast majority in my searches were actually `Error`/`NoError`, so there should only be a small number remaining

I might do, my main target right now around this is getting this linter enabled in CI. I'll track this as an issue.

**Pull request merged**